examples/qtmegachatapi/chatWindow.cpp

```diff
@@ -761,6 +762,38 @@ void ChatWindow::on_mAttachBtn_clicked()
     delete parent;
 }
 
+void ChatWindow::attachNode(QGroupBox *groupBox, mega::MegaHandle nodeHandle)
+{
+    int num = groupBox->children().size();
+    for (int i=0; i< num; i++)
+    {
+       QRadioButton *btn = (QRadioButton *) groupBox->children().at(i);
+       if (btn->isChecked())
+       {
+           if (btn->objectName() == "vc")
+           {
+               mMegaChatApi->attachVoiceMessage(mChatRoom->getChatId(), nodeHandle);
+               break;
+           }
+           else if  (btn->objectName() == "f")
+           {
+               mMegaChatApi->attachNode(mChatRoom->getChatId(), nodeHandle);
+               break;
+           }
+       }
+    }
+
+    groupBox->close();
+    if(groupBox->layout())
+    {
+        while (groupBox->layout()->count()>0)
+        {
+          delete groupBox->layout()->takeAt(0);
```

Comment:
Instead of caring about deletion of each widget, use children's hierarchy. You simply set the parent of each button and it will be automatically destroyed when the parent is destroyed